Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -600,17 +600,20 @@ private async Task<BoolResult> RestoreCheckpointStateAsync(OperationContext cont
             {
                 var latestCheckpoint = _checkpointManager.GetLatestCheckpointInfo(context);
                 var latestCheckpointAge = _clock.UtcNow - latestCheckpoint?.checkpointTime;
-                var shouldRestoreInBackground = latestCheckpointAge < _configuration.Checkpoint.RestoreCheckpointAgeThreshold;
+
+                // Only skip if this is the first restore and it is sufficiently recent
+                var shouldSkipRestore = _lastRestoreTime == default 

```

Comment:
>_lastRestoreTime [](start = 40, length = 16)

I'm not sure if this is intentional, but _lastRestoreTime is set only for the successful case. So it is possible to hit this case in a normal case (not on startup) as well.

Does it make sense to just pass 'force: true' only when we call this method from the startup?

Reply:
Added comment.

---
In reply to: [339677607](https://github.com/microsoft/BuildXL/pull/1105#discussion_r339677607) [](ancestors = 339677607)